code/Helper/Indexer.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * A collection of helper functions related to indexing.
+ */
+class Algolia_Algoliasearch_Helper_Indexer extends Mage_Core_Helper_Abstract
+{
+    private $engine = null;
+
+    /**
+     * This function will update all the requested categories and their child
+     * categories in Algolia. You can provide either a single category ID or an
+     * array of category IDs. A category ID should be either a string or an
+     * integer.
+     *
+     * @param array|string|int $updateCategoryIds
+     */
+    public function reindexSpecificCategories($updateCategoryIds)
+    {
+        $updateCategoryIds = is_array($updateCategoryIds) ? $updateCategoryIds : [$updateCategoryIds];
+
+        foreach ($updateCategoryIds as $id) {
+            $categories = Mage::getModel('catalog/category')->getCategories($id);
+
+            foreach ($categories as $category) {
+                $updateCategoryIds[] = $category->getId();
+            }
+        }
+
+        $this->getEngine()->rebuildCategoryIndex(null, $updateCategoryIds);
+    }
+
+    /**
+     * This function will update all the requested products and their parent
+     * products in Algolia. You can provide either a single product ID or an
+     * array of product IDs. A product ID should be either a string or an
+     * integer.
+     *
+     * @param array|string|int $updateProductIds
+     */
+    public function reindexSpecificProducts($updateProductIds)
+    {
+        $updateProductIds = is_array($updateProductIds) ? $updateProductIds : array($updateProductIds);
+        $productIds = $updateProductIds;
+
+        foreach ($updateProductIds as $updateProductId) {
+            if (!$this->isProductComposite($updateProductId)) {
+                $parentIds = $this->getResource()->getRelationsByChild($updateProductId);
+
+                if (!empty($parentIds)) {
+                    $productIds = array_merge($productIds, $parentIds);
+                }
+            }
+        }
+
+        if (!empty($productIds)) {
+            $this->getEngine()->removeProducts(null, $productIds);
+            $this->getEngine()->rebuildProductIndex(null, $productIds);
+        }
+    }
+
+    /**
+     * @param int $productId
+     * @return bool
+     */
+    protected function isProductComposite($productId)
+    {
+        /** @var $product Mage_Catalog_Model_Product */
+        $product = Mage::getModel('catalog/product')->load($productId);
+        return $product->isComposite();
+    }
+
+    /**
+     * @return Mage_CatalogSearch_Model_Resource_Indexer_Fulltext
+     */
+    protected function getResource()
+    {
+        return Mage::getResourceSingleton('catalogsearch/indexer_fulltext');
+    }
+
+    /**
+     * @return Algolia_Algoliasearch_Model_Resource_Engine
+     */
+    protected function getEngine()
+    {
+        if (is_null($this->engine)) {
```

Comment:
I would prefer not to set `NULL` to `$engine` variable and testing `!isset($this->engine)`, but that's only cosmetic update